### ISSUE TITLE
Ammonite 3.0.0 isn't published for Scala 3.3.3

### ISSUE THREAD

**Issue by Gedochao** [com-lihaoyi/Ammonite#1563]
```bash
scala-cli -S lts --amm --power --ammonite-version 3.0.0
# Downloading Ammonite 3.0.0
# Downloading Ammonite 3.0.0 sources
# [error]  Can't download Ammonite 3.0.0 for Scala 3.3.3.
# Ammonite with this Scala version might not yet be supported.
# Try passing a different Scala version with the -S option.
```
while it works fine with `3.0.0-M2-15-9bed9700`
```bash
scala-cli -S lts --amm --power --ammonite-version 3.0.0-M2-15-9bed9700
# Downloading Ammonite 3.0.0-M2-15-9bed9700
# Downloading Ammonite 3.0.0-M2-15-9bed9700 sources
# Loading...
# Welcome to the Ammonite Repl 3.0.0-M2-15-9bed9700 (Scala 3.3.3 Java 17)
# @
```

**Comment by lihaoyi** [comment]
This is expected because upickle does not support scala 3.3.3 due to a compiler macro issue IIRC, so both only support 3.4.2 and above

**Comment by Gedochao** [comment]
Right. So we should downgrade to 3.0.0-M2-15-9bed9700 for Scala <3.4.2. 😞 Got it.

**Comment by lihaoyi** [comment]
Once Scala 3.3.4 or 3.3.5 comes out we can add back support in Ammonite 3.x, see this pr for details https://github.com/com-lihaoyi/upickle/pull/610

**Comment by Gedochao** [comment]
Yeah... it's slightly awkward where I'd have to do special handling for <3.4.2 for the time being... or just not upgrade `ammonite` in Scala CLI until the fix is in place. 
Leaning towards the latter.

Also, iI find it a bit odd for this to be introduced in Ammonite 3.0.0 (the first stable release since 2.5.11). This is essentially removing Scala LTS support, even if it's temporary... 😕

**Comment by Gedochao** [comment]
BTW Scala 3.3.4 artifacts are up, not sure if that doesn't help already.
- https://github.com/scala/scala3/issues/21644
- https://github.com/com-lihaoyi/Ammonite/issues/1564

**Comment by lihaoyi** [comment]
Fixed in 3.0.1
